readthedocs/embed/v3/views.py

```diff
@@ -8,23 +8,37 @@
 import structlog
 from django.conf import settings
 from django.core.cache import cache
-from django.shortcuts import get_object_or_404
 from rest_framework import status
 from rest_framework.renderers import BrowsableAPIRenderer, JSONRenderer
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from selectolax.parser import HTMLParser
 
 from readthedocs.api.mixins import CDNCacheTagsMixin, EmbedAPIMixin
+from readthedocs.api.v2.permissions import IsAuthorizedToViewVersion
 from readthedocs.api.v3.permissions import HasEmbedAPIAccess
 from readthedocs.core.utils.extend import SettingsOverrideObject
 from readthedocs.embed.utils import clean_references
-from readthedocs.projects.constants import PUBLIC
 from readthedocs.storage import build_media_storage
 
 log = structlog.get_logger(__name__)
 
 
+class IsAuthorizedToGetContenFromVersion(IsAuthorizedToViewVersion):
+
+    """
+    Checks if the user from the request has permissions to get content from the version.
+
+    If the URL is from an external project, we return ``True``,
+    since we don't have a project to check for.
+    """
+
+    def has_permission(self, request, view):
+        if view.external:
```

Comment:
Should `view` be `version`?

Reply:
This class inherits from Django's REST framework `BasePermission` class, that class receives a request and a view. `external` is a method from the view (and here external means that the requested content is from an external site, this isn't an "external version")

https://github.com/readthedocs/readthedocs.org/blob/5f153d9bf5ae108ca35647a985fbe63849167eb2/readthedocs/embed/v3/views.py#L51-L55

Reply:
Once we start using type hints, I won't have to ask stuff like this :)

Thanks for explaining!!

Reply:
Big +1 on starting to use them :D